resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -422,6 +426,33 @@ private[yarn] class YarnAllocator(
     }
   }
 
+  private def handleNodesInDecommissioningState(allocateResponse: AllocateResponse): Unit = {
+    try {
+      // Some of the nodes are put in decommissioning state where RM did allocate
+      // resources on those nodes for earlier allocateResource calls, so notifying driver
+      // to put those executors in decommissioning state
+      allocateResponse.getUpdatedNodes.asScala.filter(_.getNodeState == NodeState.DECOMMISSIONING).
+        foreach(node => driverRef.send(DecommissionExecutorsOnHost(getHostAddress(node))))
+    } catch {
+      case e: Exception => logError("Sending Message to Driver to Decommission Executors" +
+        "on Decommissioning Nodes failed", e)
+    }
+  }
+
+  private def isYarnExecutorDecommissionEnabled: Boolean = {
```

Comment:
Make this a field variable (and move it to the top where fields are getting initialized)